lib/predicate/registry.js

```diff
@@ -0,0 +1,367 @@
+import path from 'path';
+import domain from 'domain';
+import stream from 'stream';
+import { errors } from 'arsenal';
+import validate from './validate';
+
+const PredicateError = require('swerrs').extend({
+    name: 'PredicateError',
+});
+
+const eventInfoSpec = {
+    // TODO: expand to conver desired event types
+    eventName: /^ObjectCreated:Put$/,
+    // TODO: what is the real bucket naming spec?
+    bucket: /^[\w\-\.]+$/,
+    // TODO: make a regexp out of the prefix and cache it
+    prefix(v) {
+        if (!v) {
+            this.prefix = '';
```

Comment:
`this` is the context object passed to validate, not `eventInfoSpec`, we use prefix for the registry key and we don't want an NPE [here](https://github.com/austo/S3/blob/master/lib/Config.js#L309)